### ISSUE TITLE
syntax error in xml report

### ISSUE THREAD

**Issue by konrad-arena** [phpmd/phpmd#590]
the content of the violation message should be escaped (or properly wrapped with CDATA), otherwise resulting XML report may be invalid (in example below there is `<p>`, but no `</p>`, so xml parsers complain)
 
```xml
<?xml version="1.0" encoding="UTF-8" ?>
<pmd version="@project.version@" timestamp="2019-01-04T11:02:41+01:00">
  <file name="…/Foo.php">
    <violation beginline="14" endline="14" rule="DuplicatedArrayKey" ruleset="Clean Code Rules" externalInfoUrl="http://phpmd.org/rules/cleancode.html#duplicatedarraykey" priority="2">
      Duplicated array key <p>, first declared at line 14.
    </violation>
  </file>
</pmd>
```
My php code is like
```php
$a = [
  '<p>' => true,
  '<p>' => true,
];
```

**Comment by kylekatarnls** [comment]
This has been fixed in 2.7.0.
